src/main/java/org/opensearch/geospatial/index/mapper/xypoint/XYPoint.java

```diff
@@ -61,4 +80,160 @@ public void resetCoords(double x, double y) {
     public Point asGeometry() {
         return new Point(getX(), getY());
     }
+
+    /**
+     * Set x and y coordinates of XYPoint if input contains WKT or coordinates.
+     *
+     * @param value  input String which needs to be parsed and validated
+     * @param ignoreZValue  boolean parameter which decides if z coordinate needs to be ignored or not
+     * @return XYPoint after setting the x and y coordinates
+     */
+    public XYPoint resetFromString(String value, final boolean ignoreZValue) {
+        Objects.requireNonNull(value, "input string which needs to be parsed should not be null");
+
+        if (value.toLowerCase(Locale.ROOT).contains(POINT_PRIMITIVE)) {
+            return resetFromWKT(value, ignoreZValue);
+        }
+        return resetFromCoordinates(value, ignoreZValue);
+    }
+
+    /**
+     * Set x and y coordinates of XYPoint if input contains coordinates.
+     *
+     * @param value  input String which contains coordinates that needs to be parsed and validated
+     * @param ignoreZValue  boolean parameter which decides if z coordinate needs to be ignored or not
+     * @return XYPoint after setting the x and y coordinates
+     * throws OpenSearchParseException
+     */
+    public XYPoint resetFromCoordinates(String value, final boolean ignoreZValue) {
```

Comment:
[Nit-pick]Why the value is not final? Also to check nulls why we are not using @NonNull annotation? 

Fix this in next PR.

Reply:
Sure, I will add final in my next PR. But, I didn't intentionally used @nonnull annotation so that in line 109 we can show an error message as well.